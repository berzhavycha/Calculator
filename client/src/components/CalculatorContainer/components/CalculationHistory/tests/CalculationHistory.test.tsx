import { render } from '@testing-library/react';
import { vi } from 'vitest'
import { CalculationHistory } from '../index';

describe('CalculationHistory component', () => {
  test('matches the snapshot', () => {
    const mockProps = {
      expression: '2 + 2',
      result: '4',
      setExpression: vi.fn(),
      setResult: vi.fn(),
    };

    const { container } = render(
      <CalculationHistory
        expression={mockProps.expression}
        result={mockProps.result}
        contentKey={"expression"}
        setExpression={mockProps.setExpression}
        setResult={mockProps.setResult}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
