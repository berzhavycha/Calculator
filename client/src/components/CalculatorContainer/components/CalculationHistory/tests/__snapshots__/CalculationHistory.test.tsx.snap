// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CalculationHistory component > matches the snapshot 1`] = `
<div>
  <div
    class="bg-gray-100 mb-6 p-2 pb-0.5 rounded-md shadow-md"
  >
    <ul>
      <li
        class="mb-2 p-1 pr-2 bg-white rounded-md shadow-md hover:bg-gray-100 text-xs cursor-pointer"
      >
        <p
          class="text-blue-500 font-bold text-right"
        >
          2 + 2
        </p>
      </li>
    </ul>
  </div>
</div>
`;
